backtracking/magic_sequence.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * @brief [Magic sequence](https://www.csplib.org/Problems/prob019/)
+ *
+ * @details
+ *
+ * @author [Jxtopher](https://github.com/jxtopher)
+ */
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <list>
+#include <numeric>
+#include <vector>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```